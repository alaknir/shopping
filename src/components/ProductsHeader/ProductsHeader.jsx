import React from "react";
import PropTypes from "prop-types";

const ProductsHeader = ({ heading, children }) => (
  <div className="row product-header">
    <div className="col-xs-8 product-header__heading ">{heading}</div>
    <div className="col-xs-4">{children}</div>
  </div>
);

ProductsHeader.propTypes = {
  heading: PropTypes.string.isRequired,
  children: PropTypes.element
};

ProductsHeader.defaultProps = {
  children: null
};

export default ProductsHeader;
