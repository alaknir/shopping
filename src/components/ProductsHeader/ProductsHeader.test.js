import React from "react";
import { mount } from "enzyme";

import ProductsHeader from "./ProductsHeader";

describe("ProductsHeader ", () => {
  it("Should render properly", () => {
    const component = <ProductsHeader heading="heading" />;
    const wrapper = mount(component);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render properly with children", () => {
    const component = (
      <ProductsHeader heading="heading">
        <div>this is test</div>
      </ProductsHeader>
    );
    const wrapper = mount(component);
    expect(wrapper).toMatchSnapshot();
  });
});
