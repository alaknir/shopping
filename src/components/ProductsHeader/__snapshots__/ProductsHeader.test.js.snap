// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsHeader  Should render properly 1`] = `
<ProductsHeader
  heading="heading"
>
  <div
    className="row product-header"
  >
    <div
      className="col-xs-8 product-header__heading "
    >
      heading
    </div>
    <div
      className="col-xs-4"
    />
  </div>
</ProductsHeader>
`;

exports[`ProductsHeader  Should render properly with children 1`] = `
<ProductsHeader
  heading="heading"
>
  <div
    className="row product-header"
  >
    <div
      className="col-xs-8 product-header__heading "
    >
      heading
    </div>
    <div
      className="col-xs-4"
    >
      <div>
        this is test
      </div>
    </div>
  </div>
</ProductsHeader>
`;
